### PR TITLE
DM-17281: Update difference imaging templates in datasets

### PR DESCRIPTION
**Pull request by mrawls**

### CODE REVIEW

File: config/apPipe.py [hunk 1]
```diff
@@ -32,8 +32,8 @@
     "y": "y",
     "VR": "g"}
 
-# Templates are deepCoaddPsfMatched
+# Templates are deepCoadds
```
Comment (kfindeisen) [inline config/apPipe.py]: Just to make sure: these are not supposed to be CompareWarps?

**Review by kfindeisen — COMMENTED** [review]
I'm seeing some worrying inconsistencies in which templates are included. There are 99 FITS files added and 98 deleted, and at least `10,1.fits` was deleted but not replaced. I'm trying to get git to make a human-friendly list of exactly what was added/deleted/replaced, but maybe you already know where the differences came from?

Also, please merge the two commits into one, since replacing the files without updating the config leaves the dataset in an invalid state.

**Comment by kfindeisen** [comment]
I investigated a bit and it turns out there's ~60 mismatched files between the old and the new versions. It almost looks like some kind of shift in the coordinates; for example, the old coadds had `10,1` through `10,11`, while the new ones have `10,3` through `10,13`.

**Comment by mrawls** [comment]
Good catch; I suspect this is due to some change in how a default skymap is constructed in the past year+. The patch indices are different but the spatial areas covered are the same. I have used these templates successfully on a full ap_pipe run of these three HiTS fields so I am confident they cover the full area. This will, however, make choosing the right templates to include in the CI mini dataset a bit less trivial.

File: config/apPipe.py (hunk 1 above)
Comment (mrawls) [reply config/apPipe.py]: Well, they are constructed with the CompareWarm algorithm, but the butler type is deep.

**Review by kfindeisen — APPROVED** [review]
Ok, if it was a full run then I have no objections.

File: config/apPipe.py (hunk 1 above)
Comment (kfindeisen) [reply config/apPipe.py]: Can you change the comment to explain the distinction? Looking at this file one would think they're plain old direct coadds (and the rest of the documentation simply describes them as "deep").

**Comment by mrawls** [comment]
This is ready for re-review now. The only error I see is unrelated and does not seem to affect the creation of output data products (calexp, diffim, etc.). Note I am running this with the weekly 2019_07, the latest master of ap_pipe (to avoid the main diaForcedSource bug), and this ticket branch for the dataset. The error is pasted below for reference.

```
apPipe FATAL: Failed on dataId={'date': '2015-02-18', 'filter': 'g', 'visit': 411371, 'hdu': 2, 'ccdnum': 60, 'object': 'Blind15A_42'}: LogicError: 
  File "src/table/BaseRecord.cc", line 85, in void lsst::afw::table::BaseRecord::assign(const lsst::afw::table::BaseRecord&, const lsst::afw::table::SchemaMapper&)
    Unequal schemas between input record and mapper. {0}
lsst::pex::exceptions::LogicError: 'Unequal schemas between input record and mapper.'

ip_diffim_DipoleFit WARN: DipoleFitPlugin not run on record 176682739670451097: edge failure
Traceback (most recent call last):
  File "/software/lsstsw/stack_20181012/stack/miniconda3-4.5.4-fcd27eb/Linux64/pipe_base/16.0-37-gc7333cb+3/python/lsst/pipe/base/cmdLineTask.py", line 388, in __call_
_
    result = self.runTask(task, dataRef, kwargs)
  File "/software/lsstsw/stack_20181012/stack/miniconda3-4.5.4-fcd27eb/Linux64/pipe_base/16.0-37-gc7333cb+3/python/lsst/pipe/base/cmdLineTask.py", line 447, in runTask
    return task.runDataRef(dataRef, **kwargs)
  File "/software/lsstsw/stack_20181012/stack/miniconda3-4.5.4-fcd27eb/Linux64/pipe_base/16.0-37-gc7333cb+3/python/lsst/pipe/base/timer.py", line 149, in wrapper
    res = func(self, *args, **keyArgs)
  File "/project/mrawls/ap_pipe/python/lsst/ap/pipe/ap_pipe.py", line 216, in runDataRef
    associationResults = self.runAssociation(calexpRef)
  File "/software/lsstsw/stack_20181012/stack/miniconda3-4.5.4-fcd27eb/Linux64/pipe_base/16.0-37-gc7333cb+3/python/lsst/pipe/base/timer.py", line 149, in wrapper
    res = func(self, *args, **keyArgs)
  File "/project/mrawls/ap_pipe/python/lsst/ap/pipe/ap_pipe.py", line 303, in runAssociation
    diffim)
  File "/software/lsstsw/stack_20181012/stack/miniconda3-4.5.4-fcd27eb/Linux64/ap_association/master-g2cd068832a+2/python/lsst/ap/association/diaForcedSource.py", line
 156, in run
    idFactory=idFactoryDiff)
  File "/software/lsstsw/stack_20181012/stack/miniconda3-4.5.4-fcd27eb/Linux64/meas_base/16.0-21-g0e05fd4+7/python/lsst/meas/base/forcedMeasurement.py", line 422, in g
enerateMeasCat
    newSource.assign(ref, self.mapper)
lsst.pex.exceptions.wrappers.LogicError: 
  File "src/table/BaseRecord.cc", line 85, in void lsst::afw::table::BaseRecord::assign(const lsst::afw::table::BaseRecord&, const lsst::afw::table::SchemaMapper&)
    Unequal schemas between input record and mapper. {0}
lsst::pex::exceptions::LogicError: 'Unequal schemas between input record and mapper.'
```

**Comment by kfindeisen** [comment]
@mrawls I just tried running `ap_pipe` (2019_08) on `visit=411371, ccdnum=60` on the old `ap_verify_hits2015` and didn't get that failure. Why do you say it's unrelated?

**Review by kfindeisen — APPROVED** [review]
Looks good.

**Pull request merged**